src/Utils/Match.ts

```diff
@@ -0,0 +1,26 @@
+export class UtilMatch {
+
+     public static mapToClosingChar =  {"'"  : "'",
+                                        "\"" : "\"", 
+                                        "<"  : ">",
+                                        ">"  : ">",
+                                        "("  : ")",
+                                        ")"  : ")",
+                                        "["  : "]",
+                                        "]"  : "]",
+                                        "{"  : "}", 
+                                        "}"  : "}",
+                                        };
```

Comment:
Writing functions instead of hashes would be easier to maintain.

```ts
const openingToClosing: {[key: string]: string} = {
    '\'' : '\'',
    '"'  : '"',
    '<'  : '>',
    '>'  : '>',
    '('  : ')',
    ')'  : ')',
    '['  : ']',
    ']'  : ']',
    '{'  : '}',
    '}'  : '}',
};
const closingToOpening: {[key: string]: string} = {};
Object.keys(openingToClosing).forEach(opening => {
    closingToOpening[openingToClosing[opening]] = opening;
});

export class UtilMatch {

     static closingToOpening(closing: string): string {
         return closingToOpening[closing];
     }

     static openingToClosing(opening: string): string {
         return openingToClosing[opening];
     }

}
```